### ISSUE TITLE
[XLIFF] ns0 littering output XML

### ISSUE THREAD

**Issue by ShadowXVII** [WeblateOrg/weblate#2526]
**Describe the bug**
Weblate is adjusting the namespaces of the XLIFF files when saving changes.

- This _sometimes_ happens. I can't seem to isolate the exact cause. i.e. approving some translations won't affect anything, but will cause error on merge.
- This is the same issue reporting as "problem 3" in #1432, potentially caused by `translate-toolkit`.
- The `<xml>` header is missing from the output file (⚠️?)
- There doesn't seem to be any other significant changes within the file except for the re-ordering of some attributes.

**Expected behavior**
No ns0: namespace.

**Screenshots**
![image](https://user-images.githubusercontent.com/446329/51883524-61f68f00-23d7-11e9-8182-526067d05287.png)

**Server configuration and status**
```
 * Weblate 3.4
 * Python 3.6.7
 * Django 2.1.5
 * Celery 4.2.1
 * celery-batches 0.2
 * six 1.12.0
 * social-auth-core 3.0.0
 * social-auth-app-django 3.1.0
 * django-appconf 1.0.2
 * translate-toolkit 2.3.1
 * translation-finder 1.0
 * Whoosh 2.7.4
 * defusedxml 0.5.0
 * Git 2.17.1
 * Pillow 5.1.0
 * python-dateutil 2.7.5
 * lxml 4.2.1
 * django-crispy-forms 1.7.2
 * django_compressor 2.2
 * djangorestframework 3.9.1
 * user-agents 1.1.0
 * jellyfish 0.7.1
 * diff-match-patch 20121119
 * pytz 2018.9
 * pyuca 1.2
 * PyYAML 3.12
 * tesserocr 2.4.0
 * Mercurial 4.5.3
 * git-svn 2.17.1
 * Database backends: django.db.backends.postgresql
 * Cache backends: avatar:FileBasedCache, default:RedisCache
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 4.15.0-1036-azure (x86_64)
```

**Comment by nijel** [comment]
Is your monolingual base file also xliff 1.2? This might be similar issue to https://github.com/WeblateOrg/weblate/issues/1181

**Comment by ShadowXVII** [comment]
Yeah, the example above is the monolingual base file, all XLIFF 1.2

**Comment by nijel** [comment]
Weblate should not really change monolingual base unless adding new strings. What commit message is there for this change?

**Comment by ShadowXVII** [comment]
Hmm. The reason it wants to edit the source is that they're all marked as "untranslated", despite being a source file...

I've been marking the source strings as "Translated" -- wish there was a bulk option for this -- I tried using download/upload to mark as translated, no luck.

Perhaps the issue is that Source should be considered 100% translated? I don't mind using it for review.

![image](https://user-images.githubusercontent.com/446329/51949957-d0445b80-2482-11e9-96e3-9c702adb8edf.png)

![image](https://user-images.githubusercontent.com/446329/51949987-f1a54780-2482-11e9-9552-fc9d33d526a5.png)

**Comment by ShadowXVII** [comment]
So, this happens when committing edits to non-base/source files too -- so bilingual files.

This was the commit it made after saving some translations -- this was adding new entries into the bilingual/destination xliff file. i.e. the en-US => en-GB file didn't contain these translations so were added by Weblate.
![image](https://user-images.githubusercontent.com/446329/51950405-852b4800-2484-11e9-8225-52bca72a3546.png)

**Comment by nijel** [comment]
Can you send me one such xliff file so that I can test it locally? It might something to do with additional namespaces, byt I don't see why it should break this way.

**Comment by nijel** [comment]
Looking at other occurence of this problem, I think it's related to xliff placeholders introduced in https://github.com/WeblateOrg/weblate/pull/2364.

**Comment by ShadowXVII** [comment]
I've prepared a sample set of files that replicates my setup. What's odd, is that i can't get the translations to save. After updating the translations for AU or NZ (100% translated), after I commit, the changes just disappear and it goes back to 0% translated?

Is this a bug in XLIFF handling? I'm on 3.4 with two patches around XLIFF source handling.

[TranslationTest.zip](https://github.com/WeblateOrg/weblate/files/2853605/TranslationTest.zip)
[XliffPatches.zip](https://github.com/WeblateOrg/weblate/files/2853609/XliffPatches.zip)

**Comment by nijel** [comment]
There turns out to be futher issues in the source only case you're using. I've fixed those in 07e759050a85bde1a55ed1a9847fbdfe3cac28a c5bdfaae872f53be47c83394a56fd112fadf8b54.

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).
